msg/CollisionObject.msg

```diff
@@ -1,5 +1,5 @@
-# A header, used for interpreting the poses
-Header header
+# A PoseStamped defining the object's origin. The shapes and subframe poses are defined in this frame
+geometry_msgs/PoseStamped origin
```

Comment:
```suggestion
# The object's origin. If set, the shapes and subframes are interpreted relative to this frame
geometry_msgs/PoseStamped pose
```

Reply:
`object.pose` reads much better than `object.origin` in my opinion. We already have a number of `*_poses` fields, so I would keep the terminology.

We might want to debate the `If set`. I would like to keep a fallback behavior for this in place to ease the transition.
My idea is to check for a valid quaternion or frame and fall back to using the first pose of a shape instead.

Secondly, I would propose to allow leaving the respective `*_poses` vector empty if only one shape is described.
It seems stupid to enforce users have to write down identities for all their objects.

Reply:
Agreed on the `pose` name and being allowed to leave the pose vectors empty

During the transition, the frame that was previously defined in `header` will need to be defined in the `PoseStamped`. This step is required either way.

Rather than maintaining logic where the first shape can replace the object's pose though, I would ask users to just switch the shape's pose with the object pose when porting to Noetic. For most people it shouldn't matter anyway, and going forward it will be a lot cleaner.

Reply:
I agree with the proposal to ask users to move the pose, **in case** a manual transition is really required.
I would propose to avoid that though.

> For most people it shouldn't matter

It sure does if all your code breaks because the header is suddenly gone. I'm sure I will still find code in 5 years I'll have to migrate. :confused: 
It also makes it really hard to write code compatible with MoveIt melodic and noetic.
This is a really annoying API change!

Thinking about this some more, I propose to retain the `Header` and make the object pose a `Pose` only.
The header should still refer to the object pose only. I understand why you made it a `PoseStamped` to begin with, but the only difference between that and the alternative is where you store a header. A comment, such as the one you already give, easily suffices to explain this.

The "only" use-cases for this new feature are multi-shape objects (very rare and somewhat broken in MoveIt for a long time) and subframes (which are very useful but not currently used a lot). I don't see this justifying breaking an 8 year old API that is *the* common API to communicate with the framework.

Reply:
I can get behind that. It will actually make this change very benign. I will update on the weekend.

In this case the pose might better be named `offset`. What do you think?